File: squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> res;
        int i = 0, j = nums.size() - 1;
        
        while(i <= j) {
            int iN = nums[i] * nums[i];
            int jN = nums[j] * nums[j]; 
            if(iN > jN) {
                res.push_back(iN);
                i++;
            } else {
                res.push_back(jN);
                j--;
            }
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

File: squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] res = new int[nums.length];
        int lo = 0; int hi = nums.length - 1;
        for (int i = nums.length - 1; i >= 0; i--) {
            if (Math.abs(nums[lo]) >= Math.abs(nums[hi])) {
                res[i] = nums[lo] * nums[lo];
                lo++;
            } else {
                res[i] = nums[hi] * nums[hi];
                hi--;
            }
        }
        return res;
    }
}